pkg/instrumentation/podmutator.go

```diff
@@ -75,6 +75,13 @@ func (pm *instPodMutator) Mutate(ctx context.Context, ns corev1.Namespace, pod c
 		return pod, nil
 	}
 
+	// If the Pod is marked for deletion we skip adding instrumentation to it because we change immutable fields (env).
+	// see https://github.com/open-telemetry/opentelemetry-operator/issues/1417.
+	if pod.DeletionTimestamp != nil {
+		logger.Info("Skipping pod instrumentation - marked for deletion")
+		return pod, nil
+	}
```

Comment:
Whether or not this fixes the specific issue, I believe this is proper behavior for the operator.  Let's move this above the `isAutoInstrumentationInjected` check.

Reply:
Sure!